File: Difficulty: Medium/Sort 0s, 1s and 2s/sort-0s-1s-and-2s.cpp
class Solution {
  public:
    void sort012(vector<int>& arr) {
        // code here
        int n = arr.size();
        // vector<int> v(n,0);
        int one=0;
        int two = 0;
        int zero = 0;
        
        
        for (int x : arr){
            if (x == 0){
                zero += 1;
            }
        
            if (x == 1){
                one += 1;
            }
            
            if (x == 2){
                two += 1;
            }
        }
        
        for (int j = 0;j<n;j++){
            if (zero != 0){
                arr[j] = 0;
                zero -= 1;
            }
            else if (one != 0){
                arr[j] = 1;
                one -= 1;
            }
            else if (two != 0){
                arr[j] = 2;
                two -= 1;
            }
        }
    
    }
};

File: Difficulty: Medium/Sort 0s, 1s and 2s/sort-0s-1s-and-2s.py
class Solution:
    def sort012(self, arr):
        # code here
        low , mid , high = 0, 0, len(arr) - 1
        
        while mid <= high:
            
            if arr[mid] == 0:
                
                arr[mid] , arr[low] = arr[low], arr[mid]
                mid += 1
                low += 1
                
            elif arr[mid] == 1:
                
                mid += 1
                
            else:
                
                arr[mid], arr[high] = arr[high] , arr[mid]
                high -= 1 
                
        
                
        